File: Golang/selectionsort.go
package Golang

import (
	"fmt"

	"github.com/klevinn/Golang-Projects/Array"
)

func SelectionSort() {
	n := len(Array.PresetArr)
	for i := 0; i < n; i++ {
		lowest := i
		for j := i + 1; j < n; j++ {
			if Array.PresetArr[j] < Array.PresetArr[lowest] {
				lowest = j
			}
		}

		store := Array.PresetArr[i]
		Array.PresetArr[i] = Array.PresetArr[lowest]
		Array.PresetArr[lowest] = store
	}

	fmt.Println("Sorted Array:", Array.PresetArr)
}

func RevSelectionSort() {
	n := len(Array.PresetArr)
	for i := 0; i < n; i++ {
		lowest := i
		for j := i + 1; j < n; j++ {
			if Array.PresetArr[j] > Array.PresetArr[lowest] {
				lowest = j
			}
		}

		store := Array.PresetArr[i]
		Array.PresetArr[i] = Array.PresetArr[lowest]
		Array.PresetArr[lowest] = store
	}

	fmt.Println("Reverse Sorted Array:", Array.PresetArr)
}

File: Python/selectionsort/selectionsort.py
# Simple CLI Selection Sort

def selectionSort(arr):
    n = len(arr)
    for i in range(n):
        lowest = i
        for j in range(i+1,n):
            if arr[j] < arr[lowest]:
                lowest = j

        arr[i], arr[lowest] = arr[lowest], arr[i]

def revSelectionSort(arr):
    n = len(arr)
    for i in range(n):
        lowest = i
        for j in range(i+1,n):
            if arr[j] > arr[lowest]:
                lowest = j

        arr[i], arr[lowest] = arr[lowest], arr[i]

def main():
    arr = [5,12,42,1,5,7,8]
    selectionSort(arr)
    print(f"The sorted array is {arr}")
    revSelectionSort(arr)
    print(f"The reverse sorted array is {arr}")

main()